app/assets/javascripts/components/contextual-guidance.js

```diff
@@ -0,0 +1,69 @@
+function ContextualGuidance ($module) {
+  if (!$module) $module = document
+  this.$module = $module
+  this.$fields = $module.querySelectorAll('[data-contextual-guidance]')
```

Comment:
In other places I've seen '[data-module="blah"]'. Is this one better do you think?

Reply:
I'm not able to use data-module for the contextual guidance as this component spans across the page so I'm using [data-contextual-guidance] to query the fields. I can use a `.js-field-with-guidance` class if we thing it can cause confusion with `[data-modules]` selectors.